### PR TITLE
[safetensors] add named groups to RE_SAFETENSORS_SHARD_FILE regex

### PR DESCRIPTION
**Pull request by mishig25**
add named groups to RE_SAFETENSORS_SHARD_FILE regex

safetensors equivalent of https://github.com/huggingface/huggingface.js/pull/621

### CODE REVIEW

File: packages/hub/src/lib/parse-safetensors-metadata.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const SAFETENSORS_INDEX_FILE = "model.safetensors.index.json";
 /// but in some situations safetensors weights have different filenames.
 export const RE_SAFETENSORS_FILE = /\.safetensors$/;
 export const RE_SAFETENSORS_INDEX_FILE = /\.safetensors\.index\.json$/;
-export const RE_SAFETENSORS_SHARD_FILE = /\d{5}-of-\d{5}\.safetensors$/;
+export const RE_SAFETENSORS_SHARD_FILE = /(?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```
Comment (coyotte508) [inline packages/hub/src/lib/parse-safetensors-metadata.ts:17]: Might as well add the "prefix"? If you're planning on extracting it

Comment (julien-c) [reply packages/hub/src/lib/parse-safetensors-metadata.ts:17]: in that case add `[_-]` between the captured prefix and the shard number

(most models on the hub use a dash at this position (it's what's generated by transformers), but not `bigscience/bloom`)

**Review by julien-c — APPROVED** [review]

File: packages/hub/src/lib/parse-safetensors-metadata.ts (hunk 1 above)
Comment (julien-c) [inline packages/hub/src/lib/parse-safetensors-metadata.ts:17]:
```suggestion
export const RE_SAFETENSORS_SHARD_FILE = /^(?<prefix>.*)[_-](?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```

Comment (coyotte508) [reply packages/hub/src/lib/parse-safetensors-metadata.ts:17]:
```suggestion
export const RE_SAFETENSORS_SHARD_FILE = /^(?<prefix>.*?)[_-]?(?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```

This is more accurate I think - it makes the `[-_]` optional but still tries to leave it out of `prefix`

Comment (mishig25) [reply packages/hub/src/lib/parse-safetensors-metadata.ts]: great idea !

Comment (julien-c) [reply packages/hub/src/lib/parse-safetensors-metadata.ts]: but that was my point, the `-_` is *not* optional

Comment (julien-c) [reply packages/hub/src/lib/parse-safetensors-metadata.ts]: we do want to enforce some level of convention

**Pull request merged**